crates/protocol/interop/src/safety.rs

```diff
@@ -19,12 +22,33 @@ pub enum SafetyLevel {
     Invalid,
 }
 
+impl FromStr for SafetyLevel {
+    type Err = SafetyLevelParseError;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s.to_lowercase().as_str() {
+            "finalized" => Ok(Self::Finalized),
+            "safe" => Ok(Self::Safe),
+            "local-safe" | "localsafe" => Ok(Self::LocalSafe),
+            "cross-unsafe" | "crossunsafe" => Ok(Self::CrossUnsafe),
```

Comment:
crossunsafe and localsafe are not valid safety level IMO
i think @emhane wanted to use serde_json::from_str in here, to keep things consistent